Providers/Profile.swift

```diff
@@ -677,34 +678,38 @@ open class BrowserProfile: Profile {
             notifySyncing(notification: NotificationProfileDidStartSyncing)
         }
 
-        fileprivate func endSyncing(_ result: Maybe<EngineResults>?) {
-            // loop through status's and fill sync state
+        fileprivate func endSyncing(_ result: SyncOperationResult) {
+            // loop through statuses and fill sync state
             syncLock.lock()
             defer { syncLock.unlock() }
             log.info("Ending all queued syncs.")
 
-            if let syncResult = result {
-                syncDisplayState = SyncStatusResolver(engineResults: syncResult).resolveResults()
+            syncDisplayState = SyncStatusResolver(engineResults: result.engineResults).resolveResults()
```

Comment:
Oddly enough I don't think the previous logic for the `None` case is ever used. `syncReducer`s terminal returns a `Deferred<Maybe<()>>` so the old `endSyncing` never got called with an optional.